### PR TITLE
Add JSON serializer

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This is WIP and targets #355.

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
I guess @neithernut or @TheNeikos have to do a pair-programming with me here, or at least check the docs with me, so I can be sure about how to implement the interface to yajl here.

**Comment by matthiasbeyer** [comment]
Note: Set serializing is implemented, but not tested, as the `ws_value_union` type does not support `ws_value_set` types by now!

**Comment by matthiasbeyer** [comment]
@neithernut I'm ready for your review here, I guess.

File: src/serialize/json/serializer.c [hunk 1]
```diff
@@ -0,0 +1,591 @@
...
+            self->buffer = NULL; // "I am ready here!"
+        } else {
+            // We must wait until we get a buffer where we can write the rest
+            // to.
+            ctx->yajl_buffer        += MIN(ctx->yajl_buffer_size, nbuf);
+            ctx->yajl_buffer_size   -= MIN(ctx->yajl_buffer_size, nbuf);
+        }
+
+        return write;
+    }
```
Comment (neithernut) [inline src/serialize/json/serializer.c:268]: This scope is unneccesary.

**Comment by neithernut** [comment]
I see nothing fishy.
ACK.

**Comment by matthiasbeyer** [comment]
If you want me to squash something, tell me. If not, this is ready for merge.

**Pull request merged**